buildconfig/CMake/WindowsNSIS.cmake

```diff
@@ -221,3 +277,31 @@ set (CPACK_NSIS_EXTRA_UNINSTALL_COMMANDS "
   RMDir \\\"$INSTDIR\\\\logs\\\"
   RMDir \\\"$INSTDIR\\\\docs\\\"
 ")
+
+# if the workbench executable was generated then add the shortcut installation and deletion everywhere
+if (_workbench_executable_full_name)
```

Comment:
I'd like some feedback for this, I am not sure it's the best way to handle when the Workbench shortcuts are installed/created, but it's the only way I could think of adding them dynamically, depending on whether the executable was generated successfully. 

The big problem here is that if the workbench executable creation failed, the shortcuts were still getting created on the Desktop - but when clicked they would just show the `executable does not exist anymore` dialog.

The `_workbench_executable_full_name` is only declared if the EXE generation was successful

Reply:
As commented above I think we should stop the packaging process with an error if the EXE generation wasn't successful. Does that "solve" this problem?

Reply:
The issue is that the shortcuts must only be added whenever `PACKAGE_WORKBENCH` is true. 

Changing it to a `FATAL_ERROR` when the executable fails to generate does not solve this, as there will be no attempt to generate if the workbench is not being packaged.

Therefore we would need to check anyway for `PACKAGE_WORKBENCH`, and append the shortcut commands dynamically. 

However, with the change to `FATAL_ERROR`, I think changing this check to `if (PACKAGE_WORKBENCH)` makes it a lot easier for the reader to understand the intent of the following code.

Reply:
This sounds sensible to me.